### ISSUE TITLE
JVM core dumps on invalid (non-Snappy) input to org.iq80.snappy.Snappy/uncompress

### ISSUE THREAD

**Issue by ptaoussanis** [dain/snappy#20]
Hi there,

I'm seeing a relatively large number of JVM core dumps (OpenJDK6+7, OracleJDK7) when speculatively trying to decompress bytes that turn out to _not_ actually be compressed.

I'd expect an exception in this case, not a core dump. Two questions:
1. Is this a bug, or are my expectations off?
2. Either way, any suggested workarounds for the moment? This implementation doesn't seem to have an equivalent of http://wiki.snappy-java.googlecode.com/hg/apidocs/org/xerial/snappy/Snappy.html#isValidCompressedBuffer(byte[]) as far as I can tell.

Example core-dump header (I can upload/email a full dump if it'd help):

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000000010396c1d2, pid=48055, tid=4352880640
#
# JRE version: 7.0_04-b21
# Java VM: Java HotSpot(TM) 64-Bit Server VM (23.0-b21 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# J  org.iq80.snappy.SnappyInternalUtils.copyLong([BI[BI)V
```

Thanks a lot for the great library and any assistance!
- Peter

**Comment by bilus** [comment]
This may be related: it also crashes when trying to decompress Snappy data with incorrect password. See https://github.com/ptaoussanis/nippy/issues/72 for example file and password combination.

**Comment by dain** [comment]
This repository has been migrated to https://github.com/airlift/aircompressor/.  Please test using the latest aircompressor and post there if it is still an issue.

**Comment by Marcono1234** [comment]
As follow-up: This should be fixed in version 0.5 and the security advisory GHSA-8wh2-6qhj-h7j9 has been published.

Though similar to dain's comment above, the advisory also recommends to migrate to https://github.com/airlift/aircompressor/.
